sdk/keyvault/arm-keyvault/README.md

```diff
@@ -1,112 +1,95 @@
-  - Copy the client ID and tenant ID from the Overview section of your app registration in Azure portal and use it in the browser sample below.
-- If you are writing a server side application,
-  - [Select a credential from `@azure/identity` based on the authentication method of your choice](https://aka.ms/azsdk/js/identity/examples)
-  - Complete the set up steps required by the credential if any.
-  - Use the credential you picked in the place of `DefaultAzureCredential` in the Node.js sample below.
+#### Using an Azure Active Directory Credential
 
-In the below samples, we pass the credential and the Azure subscription id to instantiate the client.
-Once the client is created, explore the operations on it either in your favorite editor or in our [API reference documentation](https://docs.microsoft.com/javascript/api) to get started.
+You can authenticate with Azure Active Directory using the [Azure Identity library][azure_identity]. To use the [DefaultAzureCredential][defaultazurecredential] provider shown below, or other credential providers provided with the Azure SDK, please install the `@azure/identity` package:
 
-#### nodejs - Authentication, client creation, and get vaults as an example written in JavaScript.
+```bash
+npm install @azure/identity
+```
 
-##### Sample code
+You will also need to **register a new AAD application and grant access to Azure KeyVaultManagement** by assigning the suitable role to your service principal (note: roles such as `"Owner"` will not grant the necessary permissions).
+Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables: `AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`.
+
+For more information about how to create an Azure AD Application check out [this guide](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal).
 
 ```javascript
-const { DefaultAzureCredential } = require("@azure/identity");
 const { KeyVaultManagementClient } = require("@azure/arm-keyvault");
-const subscriptionId = process.env["AZURE_SUBSCRIPTION_ID"];
-
-// Use `DefaultAzureCredential` or any other credential of your choice based on https://aka.ms/azsdk/js/identity/examples
-// Please note that you can also use credentials from the `@azure/ms-rest-nodeauth` package instead.
-const creds = new DefaultAzureCredential();
-const client = new KeyVaultManagementClient(creds, subscriptionId);
-const resourceGroupName = "testresourceGroupName";
-const vaultName = "testvaultName";
-client.vaults.get(resourceGroupName, vaultName).then((result) => {
-  console.log("The result is:");
-  console.log(result);
-}).catch((err) => {
-  console.log("An error occurred:");
-  console.error(err);
-});
+const { DefaultAzureCredential } = require("@azure/identity");
+const client = new KeyVaultManagementClient("<endpoint>", new DefaultAzureCredential());
```

Comment:
We probably need to manually change it into correct way of construct a client first